ts/components/dialog/debug/components.tsx

```diff
@@ -338,12 +284,112 @@ export const DebugActions = () => {
         >
           Open storage profile
         </SessionButton>
-        <AddDummyContactButton />
       </Flex>
     </>
   );
 };
 
+async function fetchContactsCountAndUpdate() {
+  const count = (await ContactsWrapperActions.getAll()).length;
+  if (count && Number.isFinite(count)) {
+    return count;
+  }
+  return 0;
+}
+
+function AddDummyContactButton() {
+  const [loading, setLoading] = useState(false);
+  const [addedCount, setAddedCount] = useState(0);
+  const [countToAdd, setCountToAdd] = useState(500);
+
+  const { data: contactsCount } = usePolling(
+    fetchContactsCountAndUpdate,
+    500,
+    'AddDummyContactButton'
+  );
+
+  return (
+    <div style={{ display: 'flex', alignItems: 'center' }}>
+      <SessionInput
+        autoFocus={false}
+        disableOnBlurEvent={true}
+        type="text"
+        value={`${countToAdd}`}
+        onValueChanged={(value: string) => {
+          const asNumber = toNumber(value);
+          if (Number.isFinite(asNumber)) {
+            setCountToAdd(asNumber);
+          }
+        }}
+        loading={loading}
+        maxLength={10}
+        ctaButton={
+          <SessionButton
+            onClick={async () => {
+              if (loading) {
+                return;
+              }
+              try {
+                setLoading(true);
+                setAddedCount(0);
+                const chunkSize = 10;
+                const allIndexes = Array.from({ length: countToAdd }).map((_unused, i) => i);
+                const chunks = chunk(allIndexes, chunkSize);
+                for (let chunkIndex = 0; chunkIndex < chunks.length; chunkIndex++) {
+                  // eslint-disable-next-line no-await-in-loop
+                  await Promise.all(chunks[chunkIndex].map(() => generateOneRandomContact()));
+                  setAddedCount(Math.min(chunkIndex * chunkSize, countToAdd));
+                }
+              } finally {
+                setLoading(false);
+                setAddedCount(0);
+              }
+            }}
+            disabled={loading}
+          >
+            {loading ? (
+              <>
+                {addedCount}/{countToAdd}...
+              </>
+            ) : (
+              `Add ${countToAdd} contacts (current: ${contactsCount})`
+            )}
+          </SessionButton>
+        }
+      />
+    </div>
+  );
+}
+
+export const DataGenerationActions = () => {
```

Comment:
Very nice! We are doing the same thing. I have an experimental actions component in my branch that only appears for experimental features we are working on.